nimble/host/services/audio/pacs/src/ble_audio_svc_pacs.c

```diff
@@ -0,0 +1,434 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "host/ble_hs.h"
+#include "host/ble_gatt.h"
+#include "host/ble_audio_codec.h"
+#include "services/pacs/ble_audio_svc_pacs.h"
+
+static uint32_t ble_svc_audio_pacs_sink_audio_locations;
+static uint32_t ble_svc_audio_pacs_source_audio_locations;
+static struct available_ctx {
+    uint16_t conn_handle;
+    uint16_t ble_svc_audio_pacs_avail_sink_contexts;
+    uint16_t ble_svc_audio_pacs_avail_source_contexts;
+}  ble_svc_audio_pacs_avail_contexts[MYNEWT_VAL(BLE_MAX_CONNECTIONS)] = {
+    [0 ... MYNEWT_VAL(BLE_MAX_CONNECTIONS) - 1] = {
+        .conn_handle = BLE_HS_CONN_HANDLE_NONE,
+        .ble_svc_audio_pacs_avail_sink_contexts = 0,
+        .ble_svc_audio_pacs_avail_source_contexts = 0
+    }
+};
+static uint16_t ble_svc_audio_pacs_sup_sink_contexts;
+static uint16_t ble_svc_audio_pacs_sup_source_contexts;
+
+static struct ble_gap_event_listener ble_pacs_listener;
+
+struct pac_read_cb_arg {
+    struct os_mbuf *om;
+    uint8_t pac_count;
+};
+
+static int
+ble_svc_audio_pacs_access(uint16_t conn_handle, uint16_t attr_handle,
+                          struct ble_gatt_access_ctxt *ctxt, void *arg);
+
+static const struct ble_gatt_svc_def ble_svc_audio_pacs_defs[] = {
+    { /*** Service: Published Audio Capabilities Service (PACS) */
+        .type = BLE_GATT_SVC_TYPE_PRIMARY,
+        .uuid = BLE_UUID16_DECLARE(BLE_SVC_AUDIO_PACS_UUID16),
+        .characteristics = (struct ble_gatt_chr_def[]) { {
+            .uuid = BLE_UUID16_DECLARE(BLE_SVC_AUDIO_PACS_CHR_UUID16_SINK_PAC),
+            .access_cb = ble_svc_audio_pacs_access,
+            .flags = BLE_GATT_CHR_F_READ | BLE_GATT_CHR_F_READ_ENC,
+        } , {
+            .uuid = BLE_UUID16_DECLARE(
+                BLE_SVC_AUDIO_PACS_CHR_UUID16_SINK_AUDIO_LOCATIONS),
```

Comment:
two comments here
1. As mentioned in other comment, we could have mynewt val which would enable given direction
2. We could have Mynewt_VAL to decide if this characteristic should have Notify property

Reply:
1. I'll add configs that enable/disable these characteristics
2. Initially I added notifications only for Available  Audio Contexts, as it's mandatory, but I think nothing stops us from adding notifications for the rest. One exception is PAC - as it depends on Codec module it should probably stay as is, as adding codecs would need to be calling service API, from host.

Reply:
On second thought we probably should (and can) support optional notifications or all characteristics